webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/TableColumnHeader/useTableColumnDnD.ts

```diff
@@ -26,5 +36,36 @@ export function useTableColumnDnD(model: IDatabaseDataModel, resultIndex: number
     canDrag: () => !model.isDisabled(resultIndex),
   });
 
-  return dndData;
+  const dndBox = useDNDBox({
+    canDrop(context) {
+      return (
+        context.hasValue(DATA_CONTEXT_DV_DDM, model) &&
+        context.hasValue(DATA_CONTEXT_DV_DDM_RESULT_INDEX, resultIndex) &&
+        !context.hasValue(DATA_CONTEXT_DV_DDM_RS_COLUMN_KEY, columnKey)
+      );
+    },
+    onDrop(context) {
+      const dndColumnKey = context.get(DATA_CONTEXT_DV_DDM_RS_COLUMN_KEY);
+
+      if (columnKey && dndColumnKey) {
+        resultSetViewAction.setColumnOrder(dndColumnKey, resultSetViewAction.columnIndex(columnKey));
+      }
+    },
+  });
+
+  const setRef = useCombinedRef(dndData.setTargetRef, dndBox.setRef);
+
+  let side: 'left' | 'right' | null = null;
```

Comment:
please add type Side = left' | 'right' | null;
use it here and in TableColumnDnD

Reply:
fixed